app/models/message.rb

```diff
@@ -1,58 +1,22 @@
 class NotVisibleError < RuntimeError; end
 class Message < ActiveRecord::Base
   include Diaspora::Federated::Base
-  include Diaspora::Guid
-  include Diaspora::Relayable
+  include Diaspora::Fields::Guid
+  include Diaspora::Fields::Author
 
-  xml_attr :text
-  xml_attr :created_at
-  xml_reader :diaspora_handle
-  xml_reader :conversation_guid
-
-  belongs_to :author, :class_name => 'Person'
-  belongs_to :conversation, :touch => true
+  belongs_to :conversation, touch: true
 
   delegate :name, to: :author, prefix: true
 
-  validates :text, :presence => true
-  validate :participant_of_parent_conversation
-
-  after_create do  # don't use 'after_commit' here since there is a call to 'save!'
-                   # inside, which would cause an infinite recursion
-    #sign comment as commenter
-    self.author_signature = self.sign_with_key(self.author.owner.encryption_key) if self.author.owner
-    self.save!
-    self
-  end
-
-  def diaspora_handle
-    self.author.diaspora_handle
-  end
-
-  def diaspora_handle= nh
-    self.author = Person.find_or_fetch_by_identifier(nh)
-  end
-
-  def conversation_guid
-    self.conversation.guid
-  end
+  # TODO: can be removed when messages are not relayed anymore
+  alias_attribute :parent, :conversation
 
-  def conversation_guid= guid
-    if cnv = Conversation.find_by_guid(guid)
-      self.conversation_id = cnv.id
-    end
-  end
-
-  def parent_class
-    Conversation
-  end
-
-  def parent
-    self.conversation
-  end
+  validates :text, presence: true
+  validate :participant_of_parent_conversation
 
-  def parent= parent
-    self.conversation = parent
+  def conversation_guid=(guid)
+    cnv = Conversation.find_by(guid: guid)
+    self.conversation = cnv if cnv
```

Comment:
Similar micro optimization consideration, how likely is it that we'll need the whole `Conversation` object later if this was called? Else we could save the allocation with `super if Conversation.exists?`.

Silently ignoring the assignment feels worrying too, but probably out of scope here to fix. Might be worth a TODO comment nonetheless.